src/System.Private.CoreLib/shared/System/Numerics/Vector.cs

```diff
@@ -2226,6 +2226,7 @@ public string ToString(string format, IFormatProvider formatProvider)
         /// <param name="value">The source vector.</param>
         /// <param name="factor">The scalar value.</param>
         /// <returns>The scaled vector.</returns>
+        [MethodImplAttribute(MethodImplOptions.AggressiveInlining)]
         public static Vector<T> operator *(Vector<T> value, T factor)
         {
             unchecked
```

Comment:
This will bloat the code significantly for the non-hardware accelerated case.

Should the non-hardware accelerated body be moved into `SoftwareFallback` local function, like what @tannergooding has been doing for similar cases elsewhere?

Reply:
> The non-hardware accelerated path will be always pretty slow. Trying to optimize does not seem worth it.

I agree.

There is a quite a bit of code in the S.N.Vector code (both here in `Vector<T>` and in `Vector2/3/4`) that is specializing the logic in each method for the software-fallback case to try and make it super-efficient. However, this also makes the code less maintainable and more complex overall (and for a case that most people won't be hitting).

I would be in favor of simplifying the software fallback code for these cases (which should generally just be going down the `if (Vector.IsHardwareAccelerated)` path, and only having a software fallback on the "core" methods).

Reply:
> Actually, I am wondering why this can't just be return new Vector<T>(factor) * value; for all cases. 

Good point, will change.